crates/bitcoin-da/src/spec/header.rs

```diff
@@ -25,6 +25,10 @@ impl BlockHeaderTrait for HeaderWrapper {
         BlockHashWrapper(self.header.block_hash())
     }
 
+    fn merkle_root(&self) -> Self::Hash {
+        BlockHashWrapper(BlockHash::from_raw_hash(self.header.merkle_root.into()))
+    }
+
```

Comment:
we need serve the witness commitment here (post seg-wit update merkle root)

or else it'd be impossible on the rollup to use post seg-wit transactions

in service.rs we extract that for InclusionMultiProof

Reply:
decided to do it in another issue